client/galaxy/scripts/components/RuleBuilder/SavedRulesSelector.vue

```diff
@@ -0,0 +1,74 @@
+<template>
+    <div class="btn-group dropdown">
+        <span
+            class="fas fa-history rule-builder-view-source"
+            v-bind:class="{ disabled: numOfSavedRules == 0 }"
+            v-b-tooltip.hover.bottom
+            :title="savedRulesMenu"
+            data-toggle="dropdown"
+            id="savedRulesButton"
+        ></span>
+        <div class="dropdown-menu" role="menu">
+            <a
+                class="rule-link dropdown-item saved-rule-item"
+                v-for="session in getRules()"
+                :key="session.dateTime"
+                @click="$emit('update-rules', JSON.parse(session.rule))"
+            >{{ session.dateTime }}</a>
+        </div>
+    </div>
+</template>
+
+<script>
+import Vue from "vue";
+import _l from "utils/localization";
+import BootstrapVue from "bootstrap-vue";
+Vue.use(BootstrapVue);
+export default {
+    data: function () {
+        return {
+            savedRulesMenu: _l("Recently used rules"),
+        };
+    },
+    props: {
+        builder: {
+            required: true,
+        },
+    },
+    computed: {
+        numOfSavedRules: function () {
+            return this.getRules().length;
```

Comment:
I don't think you should call methods from computed properties.

https://forum.vuejs.org/t/calling-method-in-computed-property/56448

Vue tracks these computed properties and what internal state they depend on. If a method is being called loading an external, non-reactive data source I don't think it can do that. Let me tinker with this and see if I can offer some advice.